ts/features/bonus/cgn/components/merchants/CgnMerchantsFilters.tsx

```diff
@@ -0,0 +1,201 @@
+                <Item>
+                  <Input
+                    value={address}
+                    autoFocus={true}
+                    onChangeText={setAddress}
+                    placeholderTextColor={IOColors.bluegreyLight}
+                    placeholder={I18n.t(
+                      "bonus.cgn.merchantsList.filter.addressPlaceholder"
+                    )}
+                  />
+                  <IconFont name="io-place" color={IOColors.bluegrey} />
+                </Item>
+                <View spacer />
+                <DistanceSlider
+                  value={distance}
+                  onValueChange={setDistance}
+                  disabled={address.length === 0}
+                />
+                <View spacer large />
+              </>
+            )}
+            <H2>{I18n.t("bonus.cgn.merchantsList.filter.categories")}</H2>
+            <View spacer small />
+            <FlatList
+              data={categories}
+              keyExtractor={cat => cat.type}
+              keyboardShouldPersistTaps={"handled"}
+              ItemSeparatorComponent={() => (
+                <ItemSeparatorComponent noPadded={true} />
+              )}
+              renderItem={(listItem: ListRenderItemInfo<Category>) => (
+                <CategoryCheckbox
+                  text={I18n.t(listItem.item.nameKey)}
+                  value={listItem.item.type}
+                  checked={checkedCategories.indexOf(listItem.item.type) !== -1}
+                  onPress={onCheckBoxPress}
+                  icon={listItem.item.icon}
+                />
+              )}
+            />
+            <View spacer large />
+            {props.isLocal && (
+              <>
+                <H2>{I18n.t("bonus.cgn.merchantsList.filter.orderTitle")}</H2>
+                <View spacer small />
+                <FlatList
+                  data={orders}
+                  keyExtractor={ord => ord.value}
+                  keyboardShouldPersistTaps={"handled"}
+                  ItemSeparatorComponent={() => (
+                    <ItemSeparatorComponent noPadded={true} />
+                  )}
+                  renderItem={(listItem: ListRenderItemInfo<OrderType>) => (
+                    <OrderOption
+                      text={I18n.t(listItem.item.label)}
+                      value={listItem.item.value}
+                      checked={
+                        selectedOrder.indexOf(listItem.item.value) !== -1
+                      }
+                      onPress={onOrderSelect}
+                    />
+                  )}
+                />
+              </>
+            )}
+          </View>
+        </ScrollView>
+        <FooterWithButtons
+          type={"TwoButtonsInlineHalf"}
+          leftButton={cancelButtonProps(
+            onRemoveButton,
+            I18n.t("bonus.cgn.merchantsList.filter.cta.cancel")
+          )}
+          rightButton={confirmButtonProps(
+            onConfirmButton,
```

Comment:
```suggestion
            props.onConfirm,
```
if there is no need we could avoid another function declaration